src/main/java/org/broadinstitute/barclay/help/BashTabCompletionDoclet.java

```diff
@@ -0,0 +1,533 @@
+ * For testing of help documentation generation.
+ */
+public class BashTabCompletionDoclet extends HelpDoclet {
+
+    // Barclay BashTabCompletionDoclet custom Command-line Arguments:
+
+    final private static String CALLER_SCRIPT_NAME = "-caller-script-name";
+
+    final private static String CALLER_SCRIPT_PREFIX_LEGAL_ARGS          = "-caller-pre-legal-args";
+    final private static String CALLER_SCRIPT_PREFIX_ARG_VALUE_TYPES     = "-caller-pre-arg-val-types";
+    final private static String CALLER_SCRIPT_PREFIX_MUTEX_ARGS          = "-caller-pre-mutex-args";
+    final private static String CALLER_SCRIPT_PREFIX_ALIAS_ARGS          = "-caller-pre-alias-args";
+    final private static String CALLER_SCRIPT_PREFIX_ARG_MIN_OCCURRENCES = "-caller-pre-arg-min-occurs";
+    final private static String CALLER_SCRIPT_PREFIX_ARG_MAX_OCCURRENCES = "-caller-pre-arg-max-occurs";
+
+    final private static String CALLER_SCRIPT_POSTFIX_LEGAL_ARGS          = "-caller-post-legal-args";
+    final private static String CALLER_SCRIPT_POSTFIX_ARG_VALUE_TYPES     = "-caller-post-arg-val-types";
+    final private static String CALLER_SCRIPT_POSTFIX_MUTEX_ARGS          = "-caller-post-mutex-args";
+    final private static String CALLER_SCRIPT_POSTFIX_ALIAS_ARGS          = "-caller-post-alias-args";
+    final private static String CALLER_SCRIPT_POSTFIX_ARG_MIN_OCCURRENCES = "-caller-post-arg-min-occurs";
+    final private static String CALLER_SCRIPT_POSTFIX_ARG_MAX_OCCURRENCES = "-caller-post-arg-max-occurs";
+
+    // =============================================
+
+    // Variables that are set on the command line when running javadoc:
+
+    /**
+     * Name of the executable / wrapper script that will actually invoke the java process.
+     * This wrapper script would call into the JAR and tell it which class to run.
+     */
+    private String callerScriptName = indexDefaultBaseFileName;
+
+    /**
+     * Arguments to the executable / wrapper script that come before any Java class names / tools.
+     *
+     * This is expected to be a space-delimited string with the options themselves as they should be
+     * typed by the user.
+     *
+     * This syntax is used to pass this information to directly to the bash completion script.
+     *
+     * Example: {@code "--help --info --list --inputFile --outFolder --memSize --multiplier"}
+     */
+    private String callerScriptPrefixLegalArgs       = "";
+
+    /**
+     * Types of the arguments that the executable / wrapper script is expecting before any Java class names / tools.
+     * The order of these space-delimited types should correspond to the contents of {@link #callerScriptPrefixLegalArgs}
+     *
+     * This is expected to be a space-delimited string of types.
+     *
+     * Currently accepted type values are the following (not case-sensitive):
+     *
+     *      {@code file}
+     *      {@code folder}
+     *      {@code directory}
+     *      {@code int}
+     *      {@code long}
```

Comment:
Similar to folder/directory, int/long and double/float resolve as the same thing (it's in the `_getCompletionWordList` function in the template).  This is here to make it easier for built-in types to complete in java.